src/coreclr/tools/Common/Compiler/DependencyAnalysis/Target_ARM64/ARM64Emitter.cs

```diff
@@ -150,24 +167,18 @@ public void EmitJMP(ISymbolNode symbol)
         {
             if (symbol.RepresentsIndirectionCell)
             {
-                Builder.RequireInitialPointerAlignment();
-
-                if (Builder.CountBytes % Builder.TargetPointerSize == 0)
-                {
-                    // Emit a NOP instruction to align the 64-bit reloc below.
-                    EmitNOP();
-                }
+                // Use ADRP/LDR pair to load the indirection cell address
+                // adrp x12, symbol
+                EmitADRP(Register.X12, symbol);
 
-                // ldr x12, [PC+0xc]
-                EmitLDR(Register.X12, 0xc);
+                // ldr x12, [x12, symbol page offset]
+                EmitLDR(Register.X12, Register.X12, symbol);
 
                 // ldr x12, [x12]
                 EmitLDR(Register.X12, Register.X12);
```

Comment:
```suggestion
```

Now that we can load the offset directly with adrp/ldr, we can remove this additional ldr instruction.